src/ixdat/readers/zilien.py

```diff
@@ -135,25 +148,27 @@ def read(self, path_to_file, cls=ECMSMeasurement, name=None, **kwargs):
 
         # Extract timestamp from filename on form: 2021-04-20 11_16_18 Measurement name
         file_stem = self._path_to_file.stem  # Part of filename before the extension
-        timestamp = timestamp_string_to_tstamp(
+        self._timestamp = timestamp_string_to_tstamp(
             timestamp_string=" ".join(file_stem.split(" ")[:2]),
             form=ZILIEN_TIMESTAMP_FORM,
         )
 
         # Parse metadata items
         with open(self._path_to_file, encoding="utf-8") as file_handle:
-            metadata, series_headers, column_headers = self._read_metadata(file_handle)
+            (
+                self._metadata,
+                self._series_headers,
+                self._column_headers,
+            ) = self._read_metadata(file_handle)
             file_position = file_handle.tell()
 
         # Read raw data
         with open(self._path_to_file, "rb") as file_handle:
```

Comment:
The PR is there https://github.com/ixdat/ixdat-large-test-files/pull/3.

I don't know exactly why it needs to be read as binary, because it is not my code. Probably, the `np.genfromtxt()` method prefers to read it as a binary instead of text, because there is a `file_handle.seek()` used. Maybe @KennethNielsen can explain more.

Reply:
Thanks for pointing me to the example data. @KennethNielsen , is there a simple explanation of why this is read as binary that we could put into a comment here?

Reply:
This is at the data part of the file, which means that we are passed the utf-8 encoded metadata. So I thought that I might as will cut the decode out of the picture and just hand it a binary file object. I thought it more efficient, but haven't tested. In any case, it doesn't really binary, as in tightly packed binary format, just that we don't need a special encoding.